models/models.js

```diff
@@ -6,15 +6,49 @@ exports.getAllTopics = () => {
         return result.rows;
     }))
 } 
-exports.getAllArticles = () => {
-    return db.query("SELECT articles.*, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC")
+exports.getAllArticles = ({ query }) => {
+    const queryValues = []
+    let _articlesQueryString = "SELECT articles.*, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id"
+    if (query.topic !== undefined) {
+        queryValues.push(query.topic)
+        _articlesQueryString += " WHERE topic = %L"
+    }
+    if (query.sort_by !== undefined) {
+        queryValues.push(query.sort_by)
+        _articlesQueryString += " GROUP BY articles.article_id ORDER BY %s"
+    }
+    else {
+        _articlesQueryString += " GROUP BY articles.article_id ORDER BY created_at"
+    }
+    if (query.order === "asc") {

```

Comment:
you could neaten this section of code up by destructuring your `query.order` and providing a default value 
eg `const {order = "DESC"} = query`
It would remove the need to check whether it is defined, but may make it less readable for you, so its up to you 😄